### DISCUSSION TITLE
Energy API Help

### DISCUSSION THREAD

**Discussion by dailow** [timdorr/tesla-api#569] [Q&A] [answered]
I am trying to figure out what the payload is for the new Energy API grid_import_export.  Any tips on how to get the request info?

**Comment by dailow** [comment, marked as answer]
Was able to find it in case anyone else needs it.

https://owner-api.teslamotors.com/api/1/energy_sites/{}/grid_import_export
Payload options:
"customer_preferred_export_rule": "battery_ok"
"customer_preferred_export_rule": "pv_only"
"disallow_charge_from_grid_with_solar_installed": false
"disallow_charge_from_grid_with_solar_installed": true

**Comment by blarralde** [comment]
Hey @dailow, how did you find this out? I want to try to find the payloads for the rest of the commands here: https://tesla-api.timdorr.com/energy-products/energy/commands
